otter/cloud_client.py

```diff
@@ -169,10 +212,12 @@ def got_auth((token, catalog)):
         else:
             request_ = add_bind_service(
                 catalog, service_name, region, log, request_)
-        if service_request.json_response:
-            request_ = add_json_response(request_)
         request_ = add_error_handling(
             service_request.success_pred, request_)
+        if service_request.json_response:
+            request_ = add_json_response(request_)
```

Comment:
Unfortunately, due to this clients expecting JSON error will get text error. I realize you've done this because errors can be non-JSON. I feel it is better if `add_json_response` does not error when parsing, instead returns text. This way the caller can handle JSON or text. I am not sure how good an idea this is. @radix: I remember raising this concern when you were implementing `add_json_response`. What are your thoughts?